sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,1125 @@
+    client_id
+),
+main_ping AS (
+  SELECT
+    client_id AS client_id,
+    MIN(sample_id) AS sample_id,
+    MIN(submission_timestamp_min) AS submission_timestamp,
+    ARRAY_AGG(app_build_id IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS app_build_id,
+    ARRAY_AGG(app_name IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS app_name,
+    ARRAY_AGG(vendor IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS vendor,
+    ARRAY_AGG(app_version IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS app_version,
+    ARRAY_AGG(distribution_id IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS distribution_id,
+    ARRAY_AGG(attribution.campaign IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS attribution_campaign,
+    ARRAY_AGG(attribution.content IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS attribution_content,
+    ARRAY_AGG(attribution.experiment IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS attribution_experiment,
+    ARRAY_AGG(attribution.medium IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS attribution_medium,
+    ARRAY_AGG(attribution.source IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS attribution_source,
+    ARRAY_AGG(default_search_engine_data_load_path IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS engine_data_load_path,
+    ARRAY_AGG(default_search_engine_data_name IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS engine_data_name,
+    ARRAY_AGG(default_search_engine_data_origin IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS engine_data_origin,
+    ARRAY_AGG(default_search_engine_data_submission_url IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS engine_data_submission_url,
+    ARRAY_AGG(city IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS city,
+    ARRAY_AGG(geo_subdivision1 IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS subdivision1,
+    ARRAY_AGG(normalized_channel IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS normalized_channel,
+    ARRAY_AGG(country IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS country,
+    ARRAY_AGG(os IGNORE NULLS ORDER BY submission_date ASC)[SAFE_OFFSET(0)] AS normalized_os,
+    ARRAY_AGG(normalized_os_version IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS normalized_os_version,
+    ARRAY_AGG(attribution.dltoken IGNORE NULLS ORDER BY submission_date ASC)[
+      SAFE_OFFSET(0)
+    ] AS download_token
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.clients_daily_v6`
+  WHERE
+    submission_date = @submission_date
+  GROUP BY
+    client_id
+),
+_current AS (
+  SELECT
+    client_id,
+    CAST(
+      mozfun.norm.get_earliest_value(
```

Comment:
These `earliest_value` dimensions are getting us to a place where we may have different fields from different pings. I'm not necessarily against that, but I wonder if there's a simpler way here.

Can we limit to just fields that are common between all the pings (a small number), and take _all_ the fields from that ping alone? That would be simpler to read and reason about. Was there a requirement to include all of these fields?

Reply:
For this case, the new_profile and main ping have the most data. However if the query gets the available columns from the main ping and the rest from the new_profile ping e.g. experiments, many fields will result in NULL unless we update with later arriving pings. This happens e.g. when retrieving app_name only from the main ping, which then is NULL when the client only reports other pings:
<img width="597" alt="image" src="https://github.com/mozilla/bigquery-etl/assets/30448600/6ab496c1-992b-4ece-9421-0cf014fb5d02">
I backfilled some data with this approach to `moz-fx-data-shared-prod.analysis.clients_first_seen_v2_day2`

pd/ Documenting also in the [requirements doc](https://docs.google.com/document/d/1B-zTNwschQgljij19Ez104qRV5RUC9zwjSN9tc1C8vc/edit) > "Implementation notes".

Reply:
This is what the query was looking like by using the new_profile ping for the main attributes https://console.cloud.google.com/bigquery?sq=470293616514:54f557893eb04b09a3b73d6377385bb7
For what I see, the benefit with mixing pings data is to have a more complete dataset, independent of the source ping..

Reply:
I see, that makes sense. Carry on!